homeassistant/components/media_player/plex.py

```diff
@@ -136,12 +209,61 @@ def update_devices():
                 continue
 
             if device.machineIdentifier not in plex_clients:
-                new_client = PlexClient(device, plex_sessions, update_devices,
+                new_client = PlexClient(optional_config, device, None,
+                                        plex_sessions, update_devices,
                                         update_sessions)
                 plex_clients[device.machineIdentifier] = new_client
                 new_plex_clients.append(new_client)
             else:
-                plex_clients[device.machineIdentifier].set_device(device)
+                plex_clients[device.machineIdentifier].refresh(device, None)
+
+        # add devices with a session and no client (ex. PlexConnect Apple TV's)
+        if optional_config[CONF_INCLUDE_NON_CLIENTS]:
+            for machine_identifier, session in plex_sessions.items():
+                if machine_identifier not in plex_clients:
+                    new_client = PlexClient(optional_config, None, session,
+                                            plex_sessions, update_devices,
+                                            update_sessions)
+                    plex_clients[machine_identifier] = new_client
+                    new_plex_clients.append(new_client)
+                else:
+                    plex_clients[machine_identifier].refresh(None, session)
+
+        # force devices to idle that do not have a valid session
+        for machine_identifier, client in plex_clients.items():
+            if client.session is None:
+                client.force_idle()
+
+        # add devices to dynamic groups
+        if optional_config[CONF_USE_DYNAMIC_GROUPS]:
+            active_entity_id_list = []
+            inactive_entity_id_list = []
+
+            for machine_identifier, client in plex_clients.items():
+                if client.entity_id:
+<<<<<<< HEAD
+                    # do not add if hidden
+                    client_states = hass.states.get(client.entity_id)
+                    if client_states is not None:
+                        if not client_states.attributes.get('hidden'):
+                            if client.state in [STATE_IDLE, STATE_OFF]:
+                                inactive_entity_id_list.append(
+                                    client.entity_id)
+                            else:
+                                active_entity_id_list.append(client.entity_id)
+=======
+                    if not client.hidden:
+                        if client.state in [STATE_IDLE, STATE_OFF]:
+                            inactive_entity_id_list.append(client.entity_id)
+                        else:
+                            active_entity_id_list.append(client.entity_id)
+>>>>>>> 4201005ca8ff3f7b584e5cbb9f3464f65c8ce2a9
+
+            # set groups with updated memberships
+            set_group_members(hass, GROUP_ACTIVE_DEVICES,
```

Comment:
Ignore, I pushed a new update (fix3) that works and doesn't include the junk text above